spec/factories/degrees.rb

```diff
@@ -17,7 +17,7 @@
       subject { Dttp::CodeSets::DegreeSubjects::MAPPING.keys.sample }
       institution { Dttp::CodeSets::Institutions::MAPPING.keys.sample }
       grade { Dttp::CodeSets::Grades::MAPPING.keys.sample }
-      graduation_year { (1900..Time.zone.today.year).to_a.sample }
+      graduation_year { rand((Time.zone.today.year - 50)..(Time.zone.today.year + 1)) }
```

Comment:
We want to avoid putting in "magic" numbers. Since the nuber `50` is now a constant i.e. `Degree::MAX_GRAD_YEARS` we can do the following:

```ruby
# frozen_string_literal: true

NEXT_YEAR = Time.zone.now.year.next

FactoryBot.define do
  factory :degree, class: Degree do
    association :trainee

    uk_degree_type

    trait :uk_degree_type do
      locale_code { :uk }
      uk_degree { HESA_DEGREE_TYPES.sample[2] }
    end

    trait :uk_degree_with_details do
      uk_degree_type

      subject { Dttp::CodeSets::DegreeSubjects::MAPPING.keys.sample }
      institution { Dttp::CodeSets::Institutions::MAPPING.keys.sample }
      grade { Dttp::CodeSets::Grades::MAPPING.keys.sample }
      graduation_year { rand((NEXT_YEAR - Degree::MAX_GRAD_YEARS)..(NEXT_YEAR + 1)) }
    end

    trait :non_uk_degree_type do
      locale_code { :non_uk }
      uk_degree { nil }
      non_uk_degree { NARIC_NON_UK.sample }
    end

    trait :non_uk_degree_with_details do
      non_uk_degree_type

      subject { Dttp::CodeSets::DegreeSubjects::MAPPING.keys.sample }
      country { Dttp::CodeSets::Countries::MAPPING.keys.sample }
      grade { Dttp::CodeSets::Grades::MAPPING.keys.sample }
      graduation_year { rand((NEXT_YEAR - Degree::MAX_GRAD_YEARS)..(NEXT_YEAR + 1)) }
    end

    trait :invalid_year do
      graduation_year { Time.zone.yesterday.year - Degree::MAX_GRAD_YEARS }
    end

    trait :future_year do
      graduation_year { NEXT_YEAR + 1 }
    end
  end
end

```

Reply:
cheers, the only thing i changed was removing the traits entirely, they wouldn't be needed again and we wouldn't be saving anything to db. Just declared graduation year using a let